privacyIDEA Authenticator/Token.swift

```diff
@@ -26,31 +38,83 @@ class Token: Codable {
         case label = "label"
         case counter = "counter"
         case period = "period"
+        case serial = "serial"
+        case enrollment_credential = "enrollment_credential"
+        case enrollment_url = "enrollment_url"
+        case expirationDate = "expirationDate"
+        case sslVerify = "sslVerify"
+        case state = "state"
+        case pendingAuths = "pendingAuths"
     }
+    
     // Initialize the token with the given values. Usable defaults are:
     // Digits = 6, Counter = 1 (for HOTP), Period = 30 (for TOTP), Algorithm = SHA1
-
-    init(type: String, digits: Int = 6, algorithm: String = "sha1", secret: Data, label: String, counter: Int = 1, period: Int?) {
-        self.algorithm = algorithm
-        self.secret = secret
+    init(type: String,label: String, serial: String, digits: Int = 6, algorithm: String = "sha1", secret: Data,  counter: Int = 1, period: Int?) {
         self.type = type
         self.label = label
+        self.serial = serial
+        
+        self.algorithm = algorithm
+        self.secret = secret
         self.counter = counter
-        self.digits = 6
-        // validate digits to be between 6 and 8
-        if digits <= 8 && digits >= 6 {
-            self.digits = digits
-        }
-        if digits > 8 { self.digits = 8 }
-        // validate the interval and set it
-        if type == "totp" {
-            if period != nil {
-                if period! == 30 || period! == 60 {
-                    self.period = period!
-                } else { self.period = 30 }
-            } else { self.period = 30 }
-        } else {
-            self.period = nil
+        self.digits = (digits == 6) ? 6 : 8
```

Comment:
I think the default otp size is 6 digits.
So if for any reason something like 5 digits arrive at this point, I would default to 6 digits, not to 8 digits.
This would be?

~~~~swift
self.digits = (digits == 8) ? 8 : 6
~~~~

Reply:
Yes